document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/handler/ProsecutionHandler.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.companieshouse.document.generator.prosecution.handler;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.companieshouse.api.model.prosecution.prosecutioncase.ProsecutionCaseStatusApi;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.document.generator.prosecution.ProsecutionDocumentInfoService;
+import uk.gov.companieshouse.document.generator.prosecution.ProsecutionType;
+import uk.gov.companieshouse.document.generator.prosecution.exception.DocumentInfoCreationException;
+import uk.gov.companieshouse.document.generator.prosecution.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.prosecution.exception.ProsecutionServiceException;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.ProsecutionDocument;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.Defendant;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.offence.Offence;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.prosecutioncase.ProsecutionCase;
+import uk.gov.companieshouse.document.generator.prosecution.service.ProsecutionService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Component
+public class ProsecutionHandler {
+
+    private static final Logger LOG =
+            LoggerFactory.getLogger(ProsecutionDocumentInfoService.MODULE_NAME_SPACE);
+
+    @Autowired
+    ProsecutionService prosecutionService;
+
+    /**
+     * Method to get the base prosecution information for dealing with prosecution documents
+     * Throws a HandlerException if any exceptions occurred in the service
+     * 
+     * @param resourceUri the URI for the defendant
+     * @return the ProsecutionDocument containing defendant, prosecution case and offence(s)
+     *         information
+     * @throws HandlerException
+     */
+    public ProsecutionDocument getProsecutionDocument(String resourceUri)
+            throws HandlerException {
+        try {
+            Defendant defendant = prosecutionService.getDefendant(resourceUri);
+            ProsecutionCase prosecutionCase = prosecutionService.getProsecutionCase("/" + defendant.getLinks().get("prosecution-case"));
```

Comment:
The missing prefixed slash should be fixed not appending a slash to one of the links in the code

Reply:
Done - Removed